lib/elixir_boilerplate_web/errors/helpers.ex

```diff
@@ -1,43 +1,47 @@
 defmodule ElixirBoilerplateWeb.Errors.Helpers do
-  @moduledoc """
-  Conveniences for translating and building error messages.
-  """
-  use Phoenix.HTML
+  alias Ecto.Changeset
+  alias ElixirBoilerplateWeb.Errors.View
 
   @doc """
-  Generates tag for inlined form input errors.
+  Generates a human-readable block containing all errors in a changeset
   """
-  def error_tag(form, field) do
-    Enum.map(Keyword.get_values(form.errors, field), fn error ->
-      content_tag(:span, translate_error(error))
-    end)
+  def error_messages(changeset) do
+    changeset
+    |> Changeset.traverse_errors(&translate_error/1)
+    |> convert_errors_to_html(changeset.data.__struct__)
   end
 
-  @doc """
-  Translates an error message using gettext.
-  """
-  def translate_error({msg, opts}) do
-    # When using gettext, we typically pass the strings we want
-    # to translate as a static argument:
-    #
-    #     # Translate "is invalid" in the "errors" domain
-    #     dgettext("errors", "is invalid")
-    #
-    #     # Translate the number of files with plural rules
-    #     dngettext("errors", "1 file", "%{count} files", count)
-    #
-    # Because the error messages we show in our forms and APIs
-    # are defined inside Ecto, we need to translate them dynamically.
-    # This requires us to call the Gettext module passing our gettext
-    # backend as first argument.
-    #
-    # Note we use the "errors" domain, which means translations
```

Comment:
I was about to ask why we renamed the "errors" domain to "ecto" but figured it made more sense. But since these comments were removed, i feel that we may lack the visibility of the expectations here... which is to have an "ecto.po" file in the translations.

Reply:
You’re right, I’ll add the explanation in `error_messages/1`’s `@doc` block.